app/controllers/admin/accounting/transactions_controller.rb

```diff
@@ -29,6 +29,20 @@ def create
     @loan = Loan.find(transaction_params[:project_id])
     authorize(@loan, :update?)
     @transaction = ::Accounting::Transaction.new(transaction_params)
+
+    # the txn model must be able to create managed txns before
+    # the closed books date that are imported from qb (e.g. from full fetcher)
+    # but user should not be able to create
+    # a txn before the closed book date via the Madeline interface.
+    # bc there is no way to distinguish Madeline-created from qb-imported
+    # at model level, we do this check here.
+    unless respects_closed_books_date(@transaction)
```

Comment:
I didn't love adding this logic to a controller action. I tried to figure out a way at the model validation level to distinguish between an imported qb txn and a txn that had been created via the Madeline txn form. I didn't find a way that always works (e.g. we have to be able to create 'Managed' txns before the closed book date when running the full fetcher). I also looked at limiting in the browser, but our date-picker is customized text field, rather than a :date field that has min built in, so I didn't go that route. Happy to try something with the datepicker, or something else if it you think it would be worth it.

Reply:
yeah, unfortunately by the time we get to :create, a user-created txn has qb data. I'm trying an ephemeral 'user-created' attr at Tom's suggestion, which is still not super clean but better than what I have I think.

Reply:
(in general I think it's good that it's hard to tell the diff between a user-created and qb-created txn).